cpp/src/parquet/column_reader.cc

```diff
@@ -143,6 +181,9 @@ class SerializedPageReader : public PageReader {
 
   void InitDecryption();
 
+  void DecompressPage(int compressed_len, int uncompressed_len,
+                      std::shared_ptr<Buffer>& page_buffer);
```

Comment:
This signature would be better:
```c++
std::shared_ptr<Buffer> DecompressPage(int compressed_len, int uncompressed_len, const std::shared_ptr<Buffer>& page_buffer);
```

Reply:
Thanks for the tip, I went for the following signature:

```
std::shared_ptr<Buffer> DecompressPage(int compressed_len, int uncompressed_len, const uint8_t* page_buffer);
```